presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -620,16 +624,48 @@ public void registerTableForView(Table tableReference)
         tablesForView.push(requireNonNull(tableReference, "table is null"));
     }
 
+    public void registerTableForMaterializedView(Table table)
+    {
+        requireNonNull(table, "table is null");
+        tablesForMV.put(table, tablesForMV.getOrDefault(table, 0) + 1);
```

Comment:
`tablesForMaterializedView.push(requireNonNull(tableReference, "table is null"));`

Reply:
Same, it wont work.